### PR TITLE
Support sending bytes data message

### PR DESCRIPTION
**Pull request by haifengx**
### Issue #, if available:
WorkTalkBD-1543

### Description of changes:
Support sending bytes data message

### Testing done:
* Added unit tests (due to the fact that the `data` argument in VideoClient.sendDataMessage() is a c string pointer, we are not able to directly verify the arguments in the test cases)
* E2E tested by hijacking demo app to send `[116, 101, 115, 116]` byte array, verified in JS SDK demo app the corresponding string `test` was received

#### Manual test cases (add more as needed):

- [x] Join meeting without CallKit
- [ ] Join meeting with CallKit as Incoming
- [ ] Join meeting with CallKit as Outgoing
- [x] Leave meeting
- [ ] Rejoin meeting
- [ ] See metrics
- [x] See attendee presence
- [x] Send audio
- [x] Receive audio
- [ ] Mute self
- [ ] Unmute self
- [ ] See local mute indicator when muted
- [ ] See remote mute indicator when other is muted
- [ ] Enable and disable local video
- [ ] Enable and disable remote video from remote side
- [ ] Send local video
- [ ] Pause and resume remote video
- [ ] Switch local camera
- [ ] Receive remote screen share

#### Integration validation (required before release)

- [x] Unit tests pass
- [ ] Build SDK against simulator with release options
- [ ] Build SDK against device with release options
- [x] Build SDK against simulator with debug options
- [ ] Build SDK against device with debug options
- [ ] Test Demo against simulator with SDK (debug build) in workspace
- [ ] Test Demo against device with SDK (debug build) in workspace
- [ ] Test DemoObjC against simulator with SDK (debug build) in workspace
- [ ] Test DemoObjC against device with SDK (debug build) in workspace
- [ ] Test Demo against simulator with SDK.framework (release build)
- [ ] Test Demo against device with SDK.framework (release build)
- [ ] Test DemoObjC against simulator with SDK.framework (release build)
- [ ] Test DemoObjC against device with SDK.framework (release build)

### Screenshots, if available:

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift [hunk 1]
```diff
@@ -14,22 +14,22 @@ import UIKit
...
-    private let defaultVideoClient: VideoClient
+    private let defaultVideoClient: VideoClientProtocol
 
     // This internal camera capture source is used for `startLocalVideo()` API without parameter.
     private let internalCaptureSource: DefaultCameraCaptureSource
     private var isInternalCaptureSourceRunning = true
     private let eventAnalyticsController: EventAnalyticsController
 
-    init(videoClient: VideoClient,
+    init(videoClient: VideoClientProtocol,
```
Comment (linsang21) [inline AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift:32]: Why this change?

Comment (haifengx) [reply AmazonChimeSDK/AmazonChimeSDK/internal/video/DefaultVideoClientController.swift:32]: `VideoClient` is a protocol defined in the ObjC layer and cannot be mocked in Swift. The workaround is to define an identical protocol in Swift and let the type conform to that, which enables mockingbird to generate mock classes automatically based on the protocol thus being able to stub/verify any functions being called or not called with, for example:
```
verify(videoClientMock.sendDataMessage(self.topic, data: any(), lifetimeMs: 0)).wasNeverCalled()
```
See the new [DefaultVideoClientControllerTests](https://github.com/aws/amazon-chime-sdk-ios/blob/8691784f43a4c9ab1c21ebdc5796bc7f64c2992d/AmazonChimeSDK/AmazonChimeSDKTests/internal/video/DefaultVideoClientControllerTests.swift) test added. We did that long time back for AudioClient.

**Review by linsang21 — APPROVED** [review]

**Review by ziyiz-amzn — APPROVED** [review]
